### PR TITLE
Do not allow two profiles have same name

### LINKED ISSUE #987: Profile's name should be unique
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Profile should be unique,  so we should not have two profiles with same name.
Actually, we can not create two profiles with same name and same type, but we can **create two profiles with same name and different types**, which is unreasonable.

**What you expected to happen**:
Can not create two profiles with same name, even though they have different types. 

**How to reproduce it (as minimally and precisely as possible)**:
Create a profile with name A and type B.
Then create another profile with name A and type C, this profile would be created successfully.

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version: latest
- OS (e.g. from /etc/os-release): Ubuntu 16.04.2 LTS (Xenial Xerus)
- Kernel (e.g. `uname -a`): Linux ubuntu 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ThisIsClark**
**What this PR does / why we need it**:
Profile's name should be unique, so check that whether the profile name is unique before creating a new profile.

**Which issue this PR fixes** :
 fixes #987

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/db/drivers/etcd/etcd.go [hunk 1]
```diff
@@ -1494,12 +1510,12 @@ func (c *Client) getProfileByName(ctx *c.Context, name, storageType string) (*mo
 
 // GetDefaultProfile
 func (c *Client) GetDefaultProfile(ctx *c.Context) (*model.ProfileSpec, error) {
-	return c.getProfileByName(ctx, "default", "block")
+	return c.getProfileByNameAndType(ctx, "default", "block")
```
Comment (wisererik) [inline pkg/db/drivers/etcd/etcd.go]: it's better to use constants instead of magic string

File: pkg/db/drivers/etcd/etcd.go [hunk 2]
```diff
@@ -1494,12 +1510,12 @@ func (c *Client) getProfileByName(ctx *c.Context, name, storageType string) (*mo
...
 }
 
 // GetDefaultProfileFileShare
 func (c *Client) GetDefaultProfileFileShare(ctx *c.Context) (*model.ProfileSpec, error) {
-	return c.getProfileByName(ctx, "default", "file")
+	return c.getProfileByNameAndType(ctx, "default", "file")
 }
```
Comment (wisererik) [inline pkg/db/drivers/etcd/etcd.go]: Same here

File: pkg/db/drivers/etcd/etcd.go (hunk 1 above)
Comment (kumarashit) [inline pkg/db/drivers/etcd/etcd.go]: Now the default profile created will be either default_block or default_file, so this call will always return nil. Right? Should we use default_block here

File: pkg/db/drivers/etcd/etcd.go [hunk 3]
```diff
@@ -1494,12 +1510,12 @@ func (c *Client) getProfileByName(ctx *c.Context, name, storageType string) (*mo
...
 // GetDefaultProfile
 func (c *Client) GetDefaultProfile(ctx *c.Context) (*model.ProfileSpec, error) {
-	return c.getProfileByName(ctx, "default", "block")
+	return c.getProfileByNameAndType(ctx, "default", "block")
 }
 
 // GetDefaultProfileFileShare
 func (c *Client) GetDefaultProfileFileShare(ctx *c.Context) (*model.ProfileSpec, error) {
-	return c.getProfileByName(ctx, "default", "file")
+	return c.getProfileByNameAndType(ctx, "default", "file")
```
Comment (kumarashit) [inline pkg/db/drivers/etcd/etcd.go]: Same here as commented on 1497

File: pkg/db/drivers/etcd/etcd.go (hunk 1 above)
Comment (ThisIsClark) [reply pkg/db/drivers/etcd/etcd.go]: I have changed the magic string to constant string.

File: pkg/db/drivers/etcd/etcd.go (hunk 2 above)
Comment (ThisIsClark) [reply pkg/db/drivers/etcd/etcd.go]: Same as above

File: pkg/db/drivers/etcd/etcd.go (hunk 1 above)
Comment (ThisIsClark) [reply pkg/db/drivers/etcd/etcd.go]: You are right.
I have changed this default name from "default" to "default_block"(same as the default name in install.sh)

File: pkg/db/drivers/etcd/etcd.go (hunk 3 above)
Comment (ThisIsClark) [reply pkg/db/drivers/etcd/etcd.go]: Same as above

**Pull request merged**